From 7e6d4ef02b5d8d7ae8c9dcc27a37e9f844b09b90 Mon Sep 17 00:00:00 2001
From: Jasmine Hassan <jasmine.aura@gmail.com>
Date: Sat, 15 Dec 2012 00:57:08 +0000
Subject: [plugins] only link background/xrandr to mate-desktop, not main process

Most of the plugins do not need gnome-desktop's functionality, so there is
no point in the main process linking against it.

http://git.gnome.org/browse/gnome-settings-daemon/commit/?id=ed54ceee46b6a56613c8aeb3d594f6a485400ef9
---
diff --git a/configure.ac b/configure.ac
index 2743ab6..e3d3c89 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,7 +66,6 @@ PKG_CHECK_MODULES(SETTINGS_DAEMON,
 
 PKG_CHECK_MODULES(SETTINGS_PLUGIN,
 	gtk+-2.0 >= $GTK_REQUIRED_VERSION
-	mate-desktop-2.0 >= $MATE_DESKTOP_REQUIRED_VERSION
 	gio-2.0 >= $GIO_REQUIRED_VERSION
 	dbus-glib-1 >= $DBUS_GLIB_REQUIRED_VERSION
 )
@@ -84,6 +83,11 @@ AC_PATH_PROG(GLIB_GENMARSHAL, glib-genmarshal)
 GLIB_GSETTINGS
 
 dnl ---------------------------------------------------------------------------
+dnl - Check for mate-desktop
+dnl ---------------------------------------------------------------------------
+PKG_CHECK_MODULES(MATE_DESKTOP, mate-desktop-2.0 >= $MATE_DESKTOP_REQUIRED_VERSION)
+
+dnl ---------------------------------------------------------------------------
 dnl - Check for libmatenotify
 dnl ---------------------------------------------------------------------------
 
diff --git a/plugins/background/Makefile.am b/plugins/background/Makefile.am
index a047573..01f0fc5 100644
--- a/plugins/background/Makefile.am
+++ b/plugins/background/Makefile.am
@@ -17,11 +17,13 @@ test_background_CPPFLAGS = \
 
 test_background_CFLAGS = \
 	$(SETTINGS_PLUGIN_CFLAGS)	\
+	$(MATE_DESKTOP_CFLAGS)		\
 	$(AM_CFLAGS)
 
 test_background_LDADD =		\
 	$(top_builddir)/mate-settings-daemon/libmsd-profile.la	\
 	$(SETTINGS_PLUGIN_LIBS)	\
+	$(MATE_DESKTOP_LIBS)		\
 	$(X11_LIBS) \
 	$(NULL)
 
@@ -30,8 +32,8 @@ plugin_LTLIBRARIES = \
 	$(NULL)
 
 libbackground_la_SOURCES = 		\
-	msd-background-plugin.h	\
-	msd-background-plugin.c	\
+	msd-background-plugin.h		\
+	msd-background-plugin.c		\
 	msd-background-manager.h	\
 	msd-background-manager.c	\
 	$(NULL)
@@ -44,14 +46,16 @@ libbackground_la_CPPFLAGS = \
 
 libbackground_la_CFLAGS = \
 	$(SETTINGS_PLUGIN_CFLAGS)	\
+	$(MATE_DESKTOP_CFLAGS)		\
 	$(AM_CFLAGS)
 
 libbackground_la_LDFLAGS = 		\
-	$(MSD_PLUGIN_LDFLAGS)	\
+	$(MSD_PLUGIN_LDFLAGS)		\
 	$(NULL)
 
 libbackground_la_LIBADD  = 		\
 	$(SETTINGS_PLUGIN_LIBS)		\
+	$(MATE_DESKTOP_LIBS)		\
 	$(NULL)
 
 plugin_in_files = 		\
diff --git a/plugins/xrandr/Makefile.am b/plugins/xrandr/Makefile.am
index 1a398d7..6bdfaae 100644
--- a/plugins/xrandr/Makefile.am
+++ b/plugins/xrandr/Makefile.am
@@ -55,6 +55,7 @@ libxrandr_la_CPPFLAGS =						\
 libxrandr_la_CFLAGS =			\
 	$(SETTINGS_PLUGIN_CFLAGS)	\
 	$(LIBMATENOTIFY_CFLAGS)		\
+	$(MATE_DESKTOP_CFLAGS)		\
 	$(AM_CFLAGS)
 
 libxrandr_la_LDFLAGS = 			\
@@ -62,7 +63,8 @@ libxrandr_la_LDFLAGS = 			\
 
 libxrandr_la_LIBADD  =			\
 	$(SETTINGS_PLUGIN_LIBS)		\
-	$(LIBMATENOTIFY_LIBS)
+	$(LIBMATENOTIFY_LIBS)		\
+	$(MATE_DESKTOP_LIBS)
 
 plugin_in_files =			\
 	xrandr.mate-settings-plugin.in
--
cgit v0.9.0.3-65-g4555
